### ISSUE TITLE
PHP 8.1 Deprecation notice: `strpos(): Passing null to parameter #1 ($haystack) of type string is deprecated`

### ISSUE THREAD

**Issue by jamesozzie** [google/site-kit-wp#7147]
## Bug Description

As reported by [one user in the support forums](https://wordpress.org/support/topic/php-8-1-warning-strpos-passing-null-to-parameter-1-haystack-of-type-stri/), when using PHP 8.1 the following PHP errors appear:

 
```
Deprecated | strpos(): Passing null to parameter #1 ($haystack) of type string is deprecated | +wp-includes/functions.php:7053 | 1 | Plugin: google-site-kit
Deprecated | str_replace(): Passing null to parameter #3 ($subject) of type array\|string is deprecated | +wp-includes/functions.php:2165
```

_The error as it appears on the front end of a site_

<kbd> ![image](https://github.com/google/site-kit-wp/assets/41326532/07caef9b-40cc-4aca-b754-31d5e2922dc7)

_The error as it appears from wp-admin_

<kbd> ![image](https://github.com/google/site-kit-wp/assets/41326532/54cde001-a6b2-4f6e-9d21-b32835453608)

## Steps to reproduce

1. Set up Site Kit on a new WordPress site with PHP 8.1
2. Enable error reporting

## Additional Context

The `debug.log` should be inspected to confirm the results here as not everything is visible via plugins like Query Monitor.

e.g.

```php
// in wp-config.php
define( 'WP_DEBUG', true );
define( 'WP_DEBUG_LOG', true );
```
The `wp-content/debug.log` file can be inspected easily using the Code Editor on instawp.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Site Kit should not raise PHP deprecation notices on PHP 8.1

## Implementation Brief

* Update the Splash screen to use the new special `null` screen instead of the literal
	* Merge https://github.com/google/site-kit-wp/pull/7148
* Update `REST_Routes` to not pass `null` to `stripos` here https://github.com/google/site-kit-wp/blob/f77756c7de0f0c602ce427a97efe8108763ffaf7/includes/Core/REST_API/REST_Routes.php#L69

### Test Coverage

* For the splash screen, add a simple E2E test, similar to others in the existing `screens` test (included above)

## QA Brief

* Set up Site Kit on a new WordPress site with PHP 8.1
* Enable error reporting
  * The `debug.log` should be inspected to confirm the results here as not everything is visible via plugins like Query Monitor. e.g.

```php
// in wp-config.php
define( 'WP_DEBUG', true );
define( 'WP_DEBUG_LOG', true );
```

* (The `wp-content/debug.log` file can be inspected using the Code Editor on instawp.)
* No errors should be visible on-screen or in the debug log.
* Follow the same QA steps on a site with a much lower version of PHP (eg. 5.6, the lowest we support) to ensure no errors appear as a result of this change as well. 

## Changelog entry

* Update code to stop deprecation warning appearing on sites using PHP 8.1.

**Comment by aaemnnosttv** [comment]
This should be a very low effort since most of it is already done so the estimate should almost entirely be for QA.

**Comment by tofumatt** [comment]
IB ✅

**Comment by tofumatt** [comment]
Moving this right to code review as all that was needed was a QA Brief, so I wrote that up and reviewed the PR itself 🙂

**Comment by mohitwp** [comment]
### QA Update ✅ 

- Tested on dev envionment.
- Verified on PHP version 8.1 and 7.0 using WP v6.3.
- Verified on PHP version 5.6 ,7.0 and 8.1 WP v6.2.2.
- Verified no PHP deprecation notices are showing now.


https://github.com/google/site-kit-wp/assets/94359491/6e3757a1-ba83-44f6-b022-57cdef68cd48

**Comment by mohitwp** [comment]
### QA Update ⚠️ 

Moving it back to QA from Approval.
@tofumatt I found that in front end `Deprecated: stripos(): Passing null to parameter #1 ($haystack) of type string is deprecated in /home/fusupexasa9642/web/ly-unicorn-luke.instawp.xyz/public_html/wp-content/plugins/google-site-kit/includes/Core/REST_API/REST_Routes.php on line 69` notice is appearing on PHP 8.1. 

In back end now PHP deprecated notices are not showing, but it's still showing in front end. 
Since @aaemnnosttv is not available this week.  Can you please guide weather should I create separate ticket for this or you will resolve this in follow-up PR ?

![image](https://github.com/google/site-kit-wp/assets/94359491/52a09cbd-3084-4094-b325-fff0ffafb409)

**Comment by tofumatt** [comment]
@mohitwp Looks like that's a legit bug, thanks for catching that. It's related to this issue so I've created a follow-up PR.

Turns out I didn't 100% fix the issue as I didn't quite get the IB as written so thought Evan's original PR was good to merge 😓

**Comment by techanvil** [comment]
Back to you for another pass, @mohitwp.

**Comment by mohitwp** [comment]
### QA Update ✅ 

Thanks @tofumatt @techanvil !

- Tested on main environment.
- The issue reported [above](https://github.com/google/site-kit-wp/issues/7147#issuecomment-1686193059) is now resolved.
- Verified on PHP version 8.1 and 7.0 using WP v6.3.
- Verified on PHP version 5.6 ,7.0 and 8.1 WP v6.2.2.
- Verified no PHP deprecation notices are showing now in back end and in front end related to Site Kit.

<details>


https://github.com/google/site-kit-wp/assets/94359491/23c59066-eda1-492d-9dbc-9795dc04228c


https://github.com/google/site-kit-wp/assets/94359491/d10aa7d5-85d1-4458-9256-50bfc03b9267

</details>

**Comment by sajidoon** [comment]
**Could you please replace your current function with the following one? It should solve the issue you're experiencing.**

Deprecated: str_replace(): Passing null to parameter #3 ($subject) of type array|string is deprecated in /homec/sajidhostriplex/public_html/whmtest/wp-includes/functions.php on line 2182

function wp_is_stream( $path ) {
    // Check if $path is null or empty
    if ( is_null($path) || empty($path) ) {
        // $path is null or empty, it can't be a stream.
        return false;
    }

    $scheme_separator = strpos( $path, '://' );

    if ( false === $scheme_separator ) {
        // $path isn't a stream.
        return false;
    }

    $stream = substr( $path, 0, $scheme_separator );

    return in_array( $stream, stream_get_wrappers(), true );
}


Deprecated: strpos(): Passing null to parameter #1 ($haystack) of type string is deprecated in /homec/sajidhostriplex/public_html/whmtest/wp-includes/functions.php on line 7127


function wp_normalize_path( $path ) {
    $wrapper = '';

    if ( wp_is_stream( $path ) ) {
        list( $wrapper, $path ) = explode( '://', $path, 2 );

        $wrapper .= '://';
    }

    // Check if $path is null or empty
    if ( is_null($path) || empty($path) ) {
        return $wrapper; // Return the wrapper part as is if $path is null or empty
    }

    // Standardize all paths to use '/'.
    $path = str_replace( '\\', '/', $path );

    // Replace multiple slashes down to a singular, allowing for network shares having two slashes.
    $path = preg_replace( '|(?<=.)/+|', '/', $path );

    // Windows paths should uppercase the drive letter.
    if ( ':' === substr( $path, 1, 1 ) ) {
        $path = ucfirst( $path );
    }

    return $wrapper . $path;
}

**Comment by bethanylang** [comment]
@sajidoon Thanks for getting in touch! This issue is now closed and a fix was released in Site Kit v1.108.0. If you're still experiencing issues, can you please [open a topic on our support forum](https://wordpress.org/support/plugin/google-site-kit/) and we can take a look? Thanks!

**Comment by benhassi** [comment]
<img width="1366" height="605" alt="Image" src="https://github.com/user-attachments/assets/e641bc7a-dbb7-484c-843e-3c7e4970efea" />

common problème
